internal/cmd/launcher.go

```diff
@@ -332,27 +332,27 @@ func analyze(gs *state.GlobalState, args []string) (k6deps.Dependencies, error)
 
 // isAnalysisRequired returns a boolean indicating if dependency analysis is required for the command
 func isAnalysisRequired(cmd *cobra.Command) bool {
-	switch cmd.Name() {
-	case "run":
-		if parent := cmd.Parent(); parent != nil {
-			switch parent.Name() {
-			case "k6", "cloud":
-				return true
-			}
-		}
-	case "upload":
-		if parent := cmd.Parent(); parent != nil {
-			return parent.Name() == "cloud"
-		}
-	case "cloud", "archive", "inspect":
-		if parent := cmd.Parent(); parent != nil {
-			return parent.Name() == "k6"
-		}
+	switch stringifyCommand(cmd) {
+	case "k6 run",
+		"k6 cloud",
+		"k6 cloud run",
+		"k6 cloud upload",
+		"k6 upload",
+		"k6 archive",
+		"k6 inspect":
+		return true
 	}
 
 	return false
```

Comment:
Now, I'm realizing that an improvement should be to list the cases where we wan to return false + the default section with a panic.

I'm concerned that if we add a new command, we might forget to include it in the `true` list. Since it would be automatically excluded, we wouldn't catch the issue until we ran a script that specifically required the command to resolve an extension. And I see the difficulty that we're going to remember to include the extension resolution for automated or manual tests, so there's a high chance we could release a version where this problem isn't discovered until a user encounters it.

Reply:
This is the case in reverse as well. If we add a new command and it shouldn't be analysed. This is just one more reason why we shouldn't be doing it as a wrapper but as part of the usual k6 flow and when we load the script to analyze it.

Reply:
>This is the case in reverse as well.

Yes, it is.

>This is just one more reason why we shouldn't be doing it as a wrapper but as part of the usual k6 flow and when we load the script to analyze it.

You know, I'm not against it. And I think @pablochacin is going to prioritize it now that the previous part is done.

Reply:
I do not think that we should be listing every command here. There is much greater chance we will forget something and break it, than anything else IMO. 

We do not add commands ... commonly enough for this to be a case of "this will happen next release". Actually apart from the `k6 cloud run` and co. addition I do not know when we last did this.